pkg/controllers/edgeconnect/controller.go

```diff
@@ -231,7 +241,8 @@ func (controller *Controller) updateVersionInfo(ctx context.Context, edgeConnect
 		registry.WithKeyChainSecret(&keyChainSecret),
 	)
 	if err != nil {
-		return err
+		log.Debug("updating finalizers failed", "name", edgeConnect.Name, "namespace", edgeConnect.Namespace, "error", err)
+		return errors.WithStack(err)
```

Comment:
this message is the same as another one, this function is about `versionInfo` and not finalizers